### DISCUSSION TITLE
Will sidekiq_retries_exhausted be called before a job disappears if dead: false or will it only be called before moving a job to the deadset?

### DISCUSSION THREAD

**Discussion by daugaard** [sidekiq/sidekiq#6305] [Q&A] [answered]
I am asking the question because the wiki (https://github.com/sidekiq/sidekiq/wiki/Error-Handling#configuration) says:
> After retrying so many times, Sidekiq will call the sidekiq_retries_exhausted hook on your Job if you've defined it. The hook receives the queued job hash as an argument and is called right before Sidekiq moves the job to the Dead set.

But from reading the code https://github.com/sidekiq/sidekiq/blob/e798c23643cd7d3f3b229016a87fec2bfeb7905a/lib/sidekiq/job_retry.rb#L240-L276 I get the impression that it is already called regardless and the decision whether to send it to the dead set or not happens after this has been called.

**Comment by mperham** [comment, marked as answer]
Your impression (and the code) are correct. The wiki page is awkwardly phrased.
